### DISCUSSION TITLE
Accessing current pixel color inside shader

### DISCUSSION THREAD

**Discussion by Ericlm** [pixijs/pixijs#8748] [Q&A] [answered]
Hello there! I'm a very beginner inside the world of shaders, and I'm trying to change the color of some pixels to always have a good contrast. But to do that, I need to access to the current pixel color (or the one "behind" the pixel I want to draw) to evaluate if contrast will be good enough or not. What is the best way to achieve this? Is there a way to get the current color inside the shader, or do I need to do something different?
Thank you! 😊

**Comment by SuperSodaSea** [comment, marked as answer]
I'm afraid you cannot read the current pixel color directly, WebGL don't allow this. Instead, you can do it this way:

1. Render the background to a [PIXI.RenderTexture](https://pixijs.download/v6.x/docs/PIXI.RenderTexture.html).
2. Use the texture as the shader's uniform.

Notice that you cannot use a `PIXI.RenderTexture` both as a shader uniform and as the render texture at the same time, this is also prohibit. [This link](https://gamedev.stackexchange.com/questions/10964/why-do-pixel-shaders-not-let-us-read-directly-from-the-framebuffer-or-the-depth) may help you more about why you cannot directly read the current pixel color.

**Reply by Ericlm** [reply]
Thank you for your answer!
